### PR TITLE
refactor cache

### PR DESCRIPTION
**Pull request by xithan**
* fix subcard/card_accessor issue
* don’t preselect current_set in rules editor

### CODE REVIEW

File: card/lib/card/cache.rb [hunk 1]
```diff
@@ -36,41 +36,44 @@ def renew
...
-      def reset_local
-        cache_by_class.each do |cc, cache|
-          if Card::Cache === cache
-            cache.reset_local
+      def reset_hard
+        cache_by_class.each do |_klass, cache|
+          if cache.is_a? Card::Cache
+            cache.hard.reset
+          else
+            warn "reset class #{cc}, #{cache.class} #{caller[0..8] * "\n"} ???"
```
Comment (ethn) [inline card/lib/card/cache.rb]: do you ever see these warnings?  we might be able to get rid of this test.  (ancient debugging?)

Comment (xithan) [inline card/lib/card/cache.rb]: never seen it. I remove it

File: card/lib/card/cache.rb [hunk 2]
```diff
@@ -96,127 +99,70 @@ def obj_to_key obj
...
-      @local[key]
-    end
-
-    def write_variable key, variable, value
-      key = @prefix + key
-      if @store && (object = @store.read key)
-        object.instance_variable_set "@#{ variable }", value
-        @store.write key, object
-      end
-      value
```
Comment (ethn) [inline card/lib/card/cache.rb]: I keep thinking we need something like `write_variable`, but I'm guessing we're not using it yet?

Comment (ethn) [inline card/lib/card/cache.rb]: oh! nevermind; it's in the other class.

**Pull request merged**